### PR TITLE
feat: adds search and filter to code blocks

### PR DESCRIPTION
**Pull request by kleinfreund**
## Changes

Adds a search and filter actions bar to most code blocks in the UI (the feature is available everywhere but purposefully not enabled for some code blocks). The search highlights matching lines in the displayed code and allows jumping through the lines using Enter or F3 (for reverse direction: Shift+Enter or Shift+F3). The filter (toggled on using Alt+R) filters the displayed code grep-like (sacrificing syntax highlighting). Both search and filter support a regular expression mode. The search input can be focused from anywhere on the page by pressing “/”. All other shortcuts only work within the context of a code block as to not interfere with existing browser functionality (e.g. Alt+Letter is a common menu accelerator shortcut, F3 commonly cycles through a browser's full text search, etc.).

## Screenshots

Screenshot from 2022-10-03 12-23-28

Screenshot from 2022-10-03 12-23-43

Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

### CODE REVIEW

**Comment by jakubdyszkiewicz** [comment]
1) Load XDS Config with search
2) Click on Stats
3) Go back to XDS Config
The config is not highlighted anymore even though we have text in the query.

Arrows are rendered in two different typefaces on MacOS

**Comment by kleinfreund** [comment]
@jakubdyszkiewicz I’ve turned the next/previous match button icons into `KIcon`s with chevrons which fixes that issue.

I’ve also made sure that line highlights are applied when initially rendering the component.

File: src/components/CodeBlock.vue [hunk 1]
```diff
@@ -55,30 +213,491 @@ const props = defineProps({
...
+      matchingLineNumbers.value = []
+
+      if (error instanceof Error) {
+        regExpError.value = error
+      } else {
+        throw error
+      }
+    }
+
+    console.log(`⏱ getMatchingLineNumbers: ${window.performance.now() - searchStart} ms`)
```
[the remark this answers is not in the document]
Comment (kleinfreund) [reply src/components/CodeBlock.vue:400]: Nope. I’ve just left them in in case we see other performance issues. I’ll remove them once this was approved.

**Review by lahabana — APPROVED** [review]
LGTM! Really think you should demo this at the next eng demo and we should consider making this a Kongponent I can imagine other parts of the company requiring something like this

File: src/components/CodeBlock.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (kleinfreund) [reply src/components/CodeBlock.vue]: I’ve removed them now.

**Pull request merged**